src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -1791,6 +1833,43 @@ let addBuildingBlockAt (excelIndex: int) (newBB: CompositeColumn) (table: Table)
         |> List.iteri (fun ci header -> ExcelHelper.addColumnAndRows (float (excelIndex + ci)) table header bodyValues.[ci] |> ignore)
     }
 
+type BodyCellType =
+| Term
+| Unitized
+| Text
+| Data
+
+let getSelectedCellType () =
+    Excel.run(fun context ->
+        promise {
+            let! result = tryGetActiveAnnotationTable context
+
+            match result with
+            | Result.Ok excelTable ->
+                let! _, rowIndex = getSelectedBuildingBlockCell excelTable context
+                let! arcTable = ArcTable.tryGetFromExcelTable(excelTable, context)
+                let! (arcMainColumn, _) = getArcMainColumn excelTable arcTable.Value context
+
+                if (arcMainColumn.Validate false) && rowIndex > 0 then
```

Comment:
You read the table in line 1850, the table and therefore all columns must be valid.

Reply:
fixed